libs/moment-datetime-adapter/adapter/moment-datetime-adapter.ts

```diff
@@ -0,0 +1,312 @@
+import { Inject, Injectable, InjectionToken, LOCALE_ID, Optional } from '@angular/core';
+import { DatetimeAdapter } from '@fundamental-ngx/core';
+
+import moment, { Moment, MomentFormatSpecification, MomentInput } from 'moment';
+
+function range<T>(length: number, mapFn: (index: number) => T): T[] {
+    return Array.from(new Array(length)).map((_, index) => mapFn(index));
+}
+
+export interface MomentDateTimeAdapterOptions {
+    strict?: boolean;
+    useUtc?: boolean;
+}
+
+export const MOMENT_DATE_TIME_ADAPTER_OPTIONS = new InjectionToken<MomentDateTimeAdapterOptions>(
+    'MOMENT_DATE_TIME_ADAPTER_OPTIONS', {
+        providedIn: 'root',
+        factory: MOMENT_DATE_TIME_ADAPTER_OPTIONS_FACTORY
+    }
+);
+
+export function MOMENT_DATE_TIME_ADAPTER_OPTIONS_FACTORY(): MomentDateTimeAdapterOptions {
+    return {
+        useUtc: false,
+        strict: false
+    };
+}
+
+@Injectable()
+export class MomentDatetimeAdapter extends DatetimeAdapter<Moment> {
+
+    private _localeData: {
+        firstDayOfWeek: number,
+        longMonths: string[],
+        shortMonths: string[],
+        narrowMonths: string[],
+        longDaysOfWeek: string[],
+        shortDaysOfWeek: string[],
+        narrowDaysOfWeek: string[]
+    };
+
+    constructor(
+        @Optional() @Inject(LOCALE_ID) localeId: string,
+        @Optional() @Inject(MOMENT_DATE_TIME_ADAPTER_OPTIONS) private _options?: MomentDateTimeAdapterOptions
+    ) {
+        super();
+
+        this.setLocale(localeId || moment.locale());
+    }
+
+    setLocale(locale: string): void {
+        super.setLocale(locale);
+
+        const momentLocaleData = moment.localeData(locale);
+        this._localeData = {
+            firstDayOfWeek: momentLocaleData.firstDayOfWeek(),
+            longMonths: momentLocaleData.months(),
+            shortMonths: momentLocaleData.monthsShort(),
+            narrowMonths: momentLocaleData.months().map((name: string) => name.charAt(0)),
```

Comment:
Not sure if it's reliable way
Will it return last letter in RTL mode?

Reply:
It should work :)